core/plugin.go

```diff
@@ -95,21 +99,60 @@ type SubscribedPlugin interface {
 }
 
 type RequestedPlugin struct {
-	path       string
-	checkSum   [sha256.Size]byte
-	signature  []byte
-	autoLoaded bool
-}
-
-func NewRequestedPlugin(path string) (*RequestedPlugin, error) {
-	rp := &RequestedPlugin{
-		path:       path,
-		signature:  nil,
-		autoLoaded: true,
-	}
-	err := rp.generateCheckSum()
-	if err != nil {
-		return nil, err
+	path      string
+	checkSum  [sha256.Size]byte
+	signature []byte
+}
+
+// NewRequestedPlugin returns a Requested Plugin which represents the plugin path, check sum and signature
```

Comment:
Provide details about the expected arguments (what are path, tmp, b? can they be empty or nil, etc)